taskcluster/ac_taskgraph/loader/build_config.py

```diff
@@ -124,7 +124,7 @@ def loader(kind, path, config, params, loaded_tasks):
     }
     # Filter away overridden jobs that we wouldn't build anyways to avoid ending up with
     # partial job entries.
-    overridden_jobs = {k: v for k, v in config.pop('overriden-jobs', {}).items() if affected_components is ALL_COMPONENTS or k in affected_components}
+    overridden_jobs = {k: v for k, v in config.pop('overriden-jobs', {}).items() if affected_components is ALL_COMPONENTS or k in jobs.keys()}
```

Comment:
At this stage, we've already generated jobs from the components. So, for the component`browser-engine-gecko-nightly` we now have 3 jobs: 

1. `browser-engine-gecko-nightly`
1. `nightly-browser-engine-gecko-nightly`
1. `release-browser-engine-gecko-nightly`

With the previous logic, only 1. was overridden. With the new logic, all 3 of them gets overridden.